### DISCUSSION TITLE
Vitest can't resolve import '#imports'(Nuxt3, vitest)

### DISCUSSION THREAD

**Discussion by gyofeel** [nuxt/nuxt#24719] [Questions] [answered]
Hello, I am using Vitest for testing on Nuxt3, vite env.
Due to our team's convention, the configuration uses 'autoImport: false'. So when importing Nuxt's built-in items (useRoute, nextTick, computed, etc.) we use import '#imports'.

When I run Vitest, the following error occurs:

- Error: Failed to resolve import "#imports" from "[file path]". Does the file exist?

How do I solve this? help.

Below is my vitest.config.js.

```javascript
import path from 'path';
import vue from '@vitejs/plugin-vue';

export default {
    plugins: [vue()],
    test: {
        globals: true,
        environment: 'jsdom'
    },
    resolve: {
        alias: {
            '@': path.resolve(__dirname, './src')
        }
    }
};

``` 

Thanks.

**Comment by ibeladi-mrsool** [comment]
I am having a similar issue, I opened this question for it, https://github.com/nuxt/nuxt/discussions/24766
Linking them to be bidirectional linked to each other

**Comment by gyofeel** [comment]
I registered this problem on issue tab.
https://github.com/nuxt/nuxt/issues/25021

@ibrahimBeladi

**Comment by Vittorio-QA** [comment]
The solution that worked for me:

`
defineProject({

... more code here

resolve: {
      alias: {
        "@": path.resolve(__dirname, "./src"),
        "#imports": path.resolve(__dirname, "./"),
        "~": path.resolve(__dirname, "./"),
      },
    },
}),`

Vitest finally stopped barking at nuxt imports.

**Comment by gyofeel** [comment, marked as answer]
@ibrahimBeladi , @Vittorio-QA 
Hi, guys.
I resolved this issue as below.
You can use defineVitestConfig in @nuxt/test-utils for vitest config.

- https://github.com/nuxt/test-utils/issues/700
